### PR TITLE
Switch to poetry and update some versions.

### PR DESCRIPTION
**Pull request by alunduil**
Current pre-commit errors: https://gist.github.com/alunduil/b7478070df427909517d2ac0d87a2f7c

### CODE REVIEW

File: .flake8 [hunk 1]
```diff
@@ -0,0 +1,2 @@
+[flake8]
```
Comment (Flameeyes) [inline .flake8:1]: Sigh for https://gitlab.com/pycqa/flake8/-/issues/428

File: .travis.yml [hunk 2]
```diff
@@ -1,37 +1,54 @@
...
 language: python
 python:
-    - '3.6' # EOL 2021-12-23
-    - '3.7' # EOL 2023-06-27
-    - '3.8' # EOL 2024-10
-    - 3.9-dev # RC1 2020-09-10
+    - '3.6'  # EOL 2021-12-23
+    - '3.7'  # EOL 2023-06-27
+    - '3.8'  # EOL 2024-10
+    - 3.9-dev   # EOL 2025-10
```
Comment (Flameeyes) [inline .travis.yml:7]: Doesn't this work with just '3.9' ? And adding 3.10-dev?

File: zfs/replicate/__init__.py [hunk 3]
```diff
@@ -1 +1,2 @@
+# -*- coding: utf-8 -*-
```
Comment (Flameeyes) [inline zfs/replicate/__init__.py:1]: Aren't these superfluous for Py3 code?

**Review by Flameeyes — APPROVED** [review]

File: .flake8 (hunk 1 above)
Comment (alunduil) [reply .flake8:1]: Yep, but I did look before adding this.

File: .travis.yml (hunk 2 above)
Comment (alunduil) [reply .travis.yml:7]: Nope, not yet.  If you want to play with it feel free.  I'll poke at it next time I need to update this as well.

File: zfs/replicate/__init__.py (hunk 3 above)
Comment (alunduil) [reply zfs/replicate/__init__.py:1]: Probably, but I probably went overboard with the pre-commit hooks.  We can remove them in another pull request if they become problematic or we see better evidence to drop them.

**Pull request merged**